source/leuven/services/openondemand.rst

```diff
@@ -191,136 +226,244 @@ of using any of the other partitions. There are however some limitations on the
-and only need to be done once for the set-up of each new kernel. This starts with activating your conda
-environment::
+To create any other kernel, first create a  :ref:`Python <create_python_conda_env>` or
+:ref:`R <create_r_conda_env>` Conda environment. The second step consists out of effectively
+creating the kernel in your ``$VSC_HOME/.local`` folder, as Jupyter will look for kernels in this location.
+The following commands should be excecuted from a shell (e.g. using 'Login Server Shell Access'), and only need
+to be done once for the set-up of each new kernel.
+If you already have an existing Python kernel, but your JupyterLab session freezes/craches when choosing your
+old kernel, you also need to repeat the following steps only once.
+This starts with activating your Conda environment::
 
       source activate <env_name>
 
-For Python you will need the ``ipykernel`` package installed in your conda environment. Then you create the kernel as follows::
+For Python you will need the ``ipykernel`` package installed in your Conda environment::
 
-      python -m ipykernel install --user --name '<env_name>' --display-name '<kernel_name>'
+      conda install ipykernel
+
+Then you create the kernel as follows::
+
+      python -m ipykernel install --user --env PYTHONPATH "" --name '<env_name>' --display-name '<kernel_name>'
 
 For R, you need both the ``jupyter_client`` and the ``irkernel`` package installed. With the following command you can create the kernel::
 
       Rscript -e 'IRkernel::installspec(name="<env_name>", displayname="<kernel_name>")'
 
-Once the kernel is created, you will see it in the 'Launcher' menu. You can now start working in your own customized environment.
+Once the kernel is (re)created, you will see it in the 'Launcher' menu. You can now start working in your own customized environment.
+Note that all user kernels are stored by default in ``${VSC_HOME}/.local/share/jupyter/kernels``, but we strongly advice you
+to stay away from modifying the contents of this folder, unless you are aware of the consequences.
 
-For more general information concerning JupyterLab, go to their `official documentation <https://docs.jupyter.org/en/latest/>`__.
+For more general information, please refer to the `official JupyterLab documentation`_.
 
 **Remarks:**
 
-- The start location is `$VSC_DATA`
-- At the moment, we do not support installing extensions in Jupyter Lab
+- The top-level notebook directory is ``$VSC_DATA``.
+- At the moment, we do not support installing extensions in JupyterLab.
+- One may create a customized kernel starting from a Python virtual environment, using one of Python modules.
+  This approach has interesting advantages, such as isolating a specific work environment, and also
+  installing packages into your data directory using ``pip install --prefix=${VSC_DATA}/<project-folder> <package-names>``.
+  If you are interested in this approach, contact your local HPC support for step-by-step instructions.
```

Comment:
(So this is the `Remark` I was referring to in my comments above and which I'd like to see converted in a separate paragraph with more info. IMO this is something that should also be illustrated here, referring to HPC support is not sufficient.)

Reply:
Indeed. I turned this one into a sub-section, so that it stands out.